src/Components/Selection.re

```diff
@@ -0,0 +1,75 @@
+[@deriving show({with_path: false})]
+type t = {
+  anchor: int,
+  focus: int,
+};
+
+type directionVariant =
+  | Left(int)
+  | Right(int);
```

Comment:
2 reasons:
 - I didn't know if someone will define another `direction` type and it'll brake. But this type is `Selection.direction` so we safe here.
 - My thought was that adding `..Variant` at the end will make better error since it'll be straight indication that this is variant type.

Reply:
Not entirely sure what you mean about the last point, but ordinary variants generally do produce  better error messages than polymorphic variants. The difference isn't all that significant for polymorphic variants with a small number of constructors though. It might even be better, since it will list the possibilities instead of just referring to the type name.